clio-dataaccess-model/src/main/scala/org/broadinstitute/clio/server/dataaccess/elasticsearch/ElasticsearchIndex.scala

```diff
@@ -48,6 +56,33 @@ class ElasticsearchIndex[CI <: ClioIndex](
     s"$rootDir$dir/"
   }
 
+  /**
+    * The ID of a json record in elasticsearch. By default, each record's ID consists
+    * of the concatenated values of the Key fields.
+    * Additional fields can be added to the ID using the ElasticsearchIndex constructor.
+    */
+  def getId(json: Json): String = {
+
+    def getAsString(key: String): String = {
+      json.asObject
+        .flatMap(_.apply(key))
+        .flatMap {
+          case s if s.isString =>
+            s.asString
+          case a if a.isArray || a.isObject =>
+            throw new RuntimeException("Arrays and objects cannot be used as ID fields")
+          case j => Option(j.toString())
```

Comment:
Not sure what the purpose of this is, you can call `toString` on any object (other than `null` of course).

Reply:
Calling `toString` on a JSON String results in the quotes being included in the string. Other JSON data types don't have this issue.

Reply:
`Option` is just to conform to the flatMap. `Json.asString` returns an `Option`, so we need to return an `Option` here too.

Reply:
Re-stating pet peeve about using `Option` instead of `Some` when you know the thing can't be `null`